backend/infrahub/lock.py

```diff
@@ -245,12 +248,48 @@ async def local_schema_wait(self) -> None:
         await self.get(name=LOCAL_SCHEMA_LOCK).event.wait()
 
     def global_schema_lock(self) -> InfrahubMultiLock:
-        return InfrahubMultiLock(_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_SCHEMA_LOCK])
+        return InfrahubMultiLock(lock_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_SCHEMA_LOCK])
 
     def global_graph_lock(self) -> InfrahubMultiLock:
-        return InfrahubMultiLock(_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_GRAPH_LOCK, GLOBAL_SCHEMA_LOCK])
+        return InfrahubMultiLock(lock_registry=self, locks=[LOCAL_SCHEMA_LOCK, GLOBAL_GRAPH_LOCK, GLOBAL_SCHEMA_LOCK])
 
 
 def initialize_lock(local_only: bool = False, service: Optional[InfrahubServices] = None) -> None:
     global registry  # pylint: disable=global-statement
     registry = InfrahubLockRegistry(local_only=local_only, service=service)
+
+
+def build_object_lock_name(name: str) -> str:
+    return f"global.object.{name}"
+
+
+def get_kinds_to_lock_on_object_mutation(node_schema: GeneratedBaseNodeSchema) -> list[str]:
+    """
+    Return kinds for which we want to lock during creating / updating an object of a given schema node.
+    Lock should be performed on schema kind and its generics having a uniqueness_constraint defined.
+    If a generic uniqueness constraint is the same as the node schema one,
+    it means node schema overrided this constraint, in which case we only need to lock on the generic.
+    """
+
+    schema_uc = None
+    kinds = []
+    if node_schema.uniqueness_constraints:
+        kinds.append(node_schema.kind)
+        schema_uc = node_schema.uniqueness_constraints
+
+    try:
+        generics_kinds = node_schema.inherit_from
```

Comment:
you can also use `node_schema.is_generic_schema` to determine if the node is a generic or not

Reply:
I replaced by `isinstance(node_schema, (ProfileSchema, GeneratedNodeSchema)` as `ProfileSchema` also contains `inherit_from` field